src/main/java/com/google/cloud/tools/gradle/appengine/appyaml/AppEngineAppYamlPlugin.java

```diff
@@ -124,27 +126,28 @@ private void configureExtensions() {
   }
 
   private void createStageTask() {
-    StageAppYamlTask stageTask =
-        project
-            .getTasks()
-            .create(
-                STAGE_TASK_NAME,
-                StageAppYamlTask.class,
-                stageTask1 -> {
-                  stageTask1.setGroup(APP_ENGINE_APP_YAML_TASK_GROUP);
-                  stageTask1.setDescription(
-                      "Stage an App Engine app.yaml based project for deployment");
-                  stageTask1.dependsOn(BasePlugin.ASSEMBLE_TASK_NAME);
-
-                  project.afterEvaluate(project -> stageTask1.setStagingConfig(stageExtension));
-                });
-    project
-        .getTasks()
-        .getByName(AppEngineCorePluginConfiguration.DEPLOY_TASK_NAME)
-        .dependsOn(stageTask);
-    project
-        .getTasks()
-        .getByName(AppEngineCorePluginConfiguration.DEPLOY_ALL_TASK_NAME)
-        .dependsOn(stageTask);
+    TaskContainer tasks = project.getTasks();
+    TaskProvider<StageAppYamlTask> stage =
+        tasks.register(
+            STAGE_TASK_NAME,
+            StageAppYamlTask.class,
+            stageTask -> {
+              stageTask.setGroup(APP_ENGINE_APP_YAML_TASK_GROUP);
+              stageTask.setDescription("Stage an App Engine app.yaml based project for deployment");
+              stageTask.dependsOn(BasePlugin.ASSEMBLE_TASK_NAME);
+            });
+    project.afterEvaluate(
+        project -> {
+          if (stage.isPresent()) {
```

Comment:
Actually, this check doesn't seem to work as I expected. It seems to always return true, whether the task has been actually created or not. :(